### PR TITLE
Deprecate body as unicode

### LINKED ISSUE #4546: Deprecate Response.body_as_unicode

### PR DESCRIPTION
**Pull request by nsirletti**
Fixes #4546 

My first PR on this project so advice/corrections are very welcome.

### CODE REVIEW

File: docs/topics/request-response.rst [hunk 1]
```diff
@@ -834,8 +834,6 @@ TextResponse objects
 
     .. automethod:: TextResponse.follow_all
 
-    .. method:: TextResponse.body_as_unicode()
-
         The same as :attr:`text`, but available as a method. This method is
         kept for backward compatibility; please prefer ``response.text``.
```
Comment (Gallaecio) [inline docs/topics/request-response.rst:838]: These 2 lines also need to be removed.

File: scrapy/http/response/text.py [hunk 2]
```diff
@@ -18,6 +18,7 @@
 from scrapy.http.response import Response
 from scrapy.utils.python import memoizemethod_noargs, to_unicode
 from scrapy.utils.response import get_base_url
+from scrapy.utils.decorators import deprecated
```
Comment (Gallaecio) [inline scrapy/http/response/text.py:21]: We aim for alphabetical order:

```suggestion
from scrapy.utils.decorators import deprecated
from scrapy.utils.python import memoizemethod_noargs, to_unicode
from scrapy.utils.response import get_base_url
```

**Comment by nsirletti** [comment]
Thanks a lot for the review ! I'm on it.

Now this might be a weird question but all this is very new to me.  From this point should git revert the test removal and create a new commit, right ?  As opposed to rebasing to get a cleaner log.

**Comment by elacuesta** [comment]
Thanks for the PR @nsirletti! Don't worry about the commit history, we can always squash when merging.
Regarding tests, please remember to add a check to make sure the deprecation warning is actually logged. You could find a similar case [here](https://github.com/scrapy/scrapy/blob/2.1.0/tests/test_item.py#L263-L273).

File: docs/topics/request-response.rst [hunk 3]
```diff
@@ -834,10 +834,6 @@ TextResponse objects
 
     .. automethod:: TextResponse.follow_all
```
Comment (Gallaecio) [inline docs/topics/request-response.rst:836]: This is quite the nitpick, but you should remove one more empty line, there are 3 now and there should be only 2 :slightly_smiling_face:

File: scrapy/http/response/text.py [hunk 4]
```diff
@@ -5,6 +5,8 @@
 See documentation in docs/topics/request-response.rst
 """
 
+import warnings
+
```
Comment (Gallaecio) [inline scrapy/http/response/text.py:9]: We don’t leave an empty line in these cases.

File: scrapy/http/response/text.py [hunk 5]
```diff
@@ -14,6 +16,8 @@
                             http_content_type_encoding, resolve_encoding)
 from w3lib.html import strip_html5_whitespace
 
+
```
Comment (Gallaecio) [inline scrapy/http/response/text.py:19]: There should be a single empty here.

File: docs/topics/request-response.rst (hunk 3 above)
Comment (nsirletti) [reply docs/topics/request-response.rst:836]: Well it's minute but important.  Thanks for the remark I'll try and be more thorough next time !

**Comment by Gallaecio** [comment]
Thanks!

**Pull request merged**